### DISCUSSION TITLE
Question about sync_list

### DISCUSSION THREAD

**Discussion by JessiBaughman** [abraunegg/onedrive#1920] [Q&A] [answered]
I noticed that when syncing, onedrive scans all the subfolders in an excluded directory I have in sync_list.
The sync_list contains the line:
`!mrc/mrc_service/aArchive`

And when I sync I get lines like:
`Skipping item - excluded by sync_list config: mrc/mrc_service/aArchive/TA`

The issue is that this being an archive folder, it contains a lot of folders and scanning each one causes the sync to last about 30 minutes. Is this normal for it to scan the subdirectories of an excluded directory or am I missing something? Do I need a trailing slash at the end of the sync_list line? I'm using onedrive version 2.14.16-3-gf26f03b on RHEL 8.5.

**Comment by abraunegg** [comment, marked as answer]
@JessiBaughman 
The answer is rather complex.

When performing a *new* sync for the first time, or in a 'resync' scenario, there is no local cache to articulate what is in-sync, what is not in sync. When the client 'syncs' in this manner, it requests from OneDrive all objects, as the OneDrive API does not permit exclusions in the the object query when tracking what changes have occurred on OneDrive (this in my mind is an API capability gap).

When the cache is fully populated, this cache also details what was the last change record processed by the client as provided - thus, we query just from the last change set point.

Now - if you change a config item that necessitates a 'resync' we go back to the above scenario, and have to go through all the OneDrive items again.

So - when OneDrive sends all the items stored online, in order, these are then 'processed' and checked against the required configuration - this is why you see `Skipping item - excluded by sync_list config: mrc/mrc_service/aArchive/TA` in the application logs.

When the application turns around and looks at local changes, the 'parent' path should be triggering an exclude of all child objects.

Hopefully this explains what you are seeing.
